tests/integration/ha/test_ha.py

```diff
@@ -411,6 +426,54 @@ async def test_freeze_db_process_node_with_elected_cm(
     await assert_continuous_writes_consistency(ops_test, c_writes, app)
 
 
+@pytest.mark.abort_on_fail
+async def test_full_cluster_crash(
+    ops_test: OpsTest,
+    c_writes: ContinuousWrites,
+    c_balanced_writes_runner,
+    reset_service_restart_delay,
+) -> None:
+    """Check cluster can operate normally after all nodes down at same time and come back up."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    # update all units to have a new RESTART_DELAY. Modifying the Restart delay to 3 minutes
+    # should ensure enough time for all replicas to be down at the same time.
+    for unit_id in get_application_unit_ids(ops_test, app):
+        await update_restart_delay(ops_test, unit_id, RESTART_DELAY)
+
+    # kill all units simultaneously
+    await asyncio.gather(
+        *[
+            send_kill_signal_to_process(ops_test, app, unit_id, signal="SIGKILL")
+            for unit_id in get_application_unit_ids(ops_test, app)
+        ]
+    )
+
+    # check that all units being down at the same time.
+    assert await all_processes_down(ops_test, app), "Not all units down at the same time."
+
+    # sleep for restart delay + 10 secs max for the election time + node start
+    time.sleep(RESTART_DELAY + 10)
```

Comment:
maybe as separate PR later we can run `reset_service_restart_delay` here and do `time.sleep(ORIGINAL_RESTART_DELAY + 10)` instead.
it will speed up the test!

Reply:
Good idea, thank you!